osu.Game/Screens/Play/PlayerLoader.cs

```diff
@@ -89,6 +89,12 @@ protected override bool OnExiting(Screen next)
         {
             Content.ScaleTo(0.7f, 150, EasingTypes.InQuint);
             FadeOut(150);
+
+            //this is required to clean up the InputManager in Player.
+            //can be removed once we solve that one.
+            if (player != null && player.LoadState != LoadState.Alive)
+                player.Dispose();

```

Comment:
Who holds a reference to player after playerloader exits that would prevent it from being cleaned up by the GC?